security/oidc.go

```diff
@@ -125,17 +126,46 @@ func (c *OIDCConfig) callbackHandler(w http.ResponseWriter, r *http.Request) { /
 		http.Redirect(w, r, "/", http.StatusFound)
 		return
 	}
-	for _, subject := range c.AllowedSubjects {
-		if strings.ToLower(subject) == strings.ToLower(idToken.Subject) {
-			c.setSessionCookie(w, idToken)
-			http.Redirect(w, r, "/", http.StatusFound)
-			return
-		}
+
+	var claimsMap map[string]any
+	if err := idToken.Claims(&claimsMap); err != nil {
+		http.Error(w, "failed to parse claims", http.StatusBadRequest)
+		return
+	}
+
+	if c.isAuthorized(idToken.Subject, claimsMap) {
+		c.setSessionCookie(w, idToken)
+		http.Redirect(w, r, "/", http.StatusFound)
+		return
 	}
-	logr.Debugf("[security.callbackHandler] Subject %s is not in the list of allowed subjects", idToken.Subject)
 	http.Redirect(w, r, "/?error=access_denied", http.StatusFound)
 }
 
+func (c *OIDCConfig) isAuthorized(subject string, claimsMap map[string]any) bool {
+	claimToCheck := c.ClaimToCheck
+	if len(claimToCheck) > 0 {
```

Comment:
Instead of doing it like this, extract the claim to compare (sub, or the custom one), and then iterate over the list of allowed subjects. It makes the code easier to read.